### PR TITLE
Feat/hpl visibility in hop

### PR DESCRIPTION
**Pull request by ochaouachi**
In this PR, We fixed the issue of "a user can view other user's pipeline in hop".
The old implementation is : 
1. When user clicks on view pipeline, the pipeline is downloaded from hop file system
2. After finishing the work in hop, the pipeline is re-uploaded to minio and deleted from minio

The issue seen when two users log in simultaneously.

**The fix:**
Now hop reads directly from minio via FTP and the specific pipeline is forced to be open in hop

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -446,6 +452,9 @@ services:
       HOP_SERVER_PASS: ${HOP_SERVER_PASS}
       HOP_SERVER_PORT: 8080
       HOP_SERVER_HOSTNAME: 0.0.0.0
+      HOP_MINIO_ACCES: minio
+      HOP_MINIO_SECRET: miniorepan
+
```
Comment (AhmAmri) [inline docker-compose.yml:457]: why exposing this credentials? anyone can login with these credentials to minio and delete files

File: backend/pipeline/validator.py [hunk 2]
```diff
@@ -1,12 +1,15 @@
 import xml.etree.ElementTree as ET
+from utils.minio import client
 
 from .rules.parquet_file_output_rule import ParquetFileOutputRule
 # A XML Schema validation check should be implemented in the future to ensure that the XML data is valid
-def check_pipeline_validity(name):
+def check_pipeline_validity(name, user_id=None):
```
Comment (AhmAmri) [inline backend/pipeline/validator.py:6]: do we need the user_id here?

File: backend/pipeline/validator.py [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
     valid_pipeline = False
     check_text = "ValidationFailed"
+    # Download the pipeline in /tmp
+    f = client.fget_object("pipelines", f"/pipelines-created/{user_id}/{name}.hpl", file_path=f"/tmp/{name}.hpl")
     # Read the .hpl file
-    with open(f"/hop/pipelines/{name}.hpl", "r") as file:
+    with open(f"/tmp/{name}.hpl", "r") as file:
```
Comment (AhmAmri) [inline backend/pipeline/validator.py:12]: here we need to use also the same improvement we did at other places. When downloading the file from minio, we first need to wait and do retry check that file is fully available on local  before trying to open it

File: docker-compose.yml (hunk 1 above)
Comment (ochaouachi) [reply docker-compose.yml:457]: Yes I Forgot this when I was testing

File: backend/pipeline/validator.py (hunk 2 above)
Comment (ochaouachi) [reply backend/pipeline/validator.py:6]: We need this to browse minio, because the file won't be saved in local file system

**Pull request merged**